qcodes/instrument_drivers/Keysight/keysight_34980a_submodules.py

```diff
@@ -0,0 +1,185 @@
+
+    def to_channel_list(
+            self,
+            paths: List[Tuple[int, int]],
+            wiring_config: Optional[str] = None
+    ) -> str:
+        """
+        convert the (row, column) pair to a 4-digit channel number 'sxxx', where
+        s is the slot number, xxx is generated from the numbering function.
+        This may be different for different modules.
+
+        Args:
+            paths: list of channels to connect [(r1, c1), (r2, c2), (r3, c3)]
+            wiring_config (str): for 1-wire matrices, values are 'MH', 'ML';
+                                 for 2-wire matrices, values are 'M1H', 'M2H',
+                                 'M1L', 'M2L'
+
+        Returns:
+            in the format of '(@sxxx, sxxx, sxxx, sxxx)', where sxxx is a
+            4-digit channel number
+        """
+        raise NotImplementedError("Please subclass this")
+
+    def is_open(self, row: int, column: int) -> bool:
+        """
+        to check if a channel is open/disconnected
+
+        Args:
+            row (int): row number
+            column (int): column number
+
+        Returns:
+            True if the channel is open/disconnected
+            False if it's closed/connected.
+        """
+        self.validate_value(row, column)
+        channel = self.to_channel_list([(row, column)])
+        message = self.ask(f'ROUT:OPEN? {channel}')
+        return bool(int(message))
+
+    def is_closed(self, row: int, column: int) -> bool:
+        """
+        to check if a channel is closed/connected
+
+        Args:
+            row (int): row number
+            column (int): column number
+
+        Returns:
+            True if the channel is closed/connected
+            False if it's open/disconnected.
+        """
+        self.validate_value(row, column)
+        channel = self.to_channel_list([(row, column)])
+        message = self.ask(f'ROUT:CLOSe? {channel}')
+        return bool(int(message))
+
+    def connect(self, row: int, column: int) -> None:
+        """
+        to connect/close the specified channels
+
+        Args:
+            row (int): row number
+            column (int): column number
+        """
+        self.validate_value(row, column)
+        channel = self.to_channel_list([(row, column)])
+        self.write(f'ROUT:CLOSe {channel}')
+
+    def disconnect(self, row: int, column: int) -> None:
+        """
+        to disconnect/open the specified channels
+
+        Args:
+            row (int): row number
+            column (int): column number
+        """
+        self.validate_value(row, column)
+        channel = self.to_channel_list([(row, column)])
+        self.write(f'ROUT:OPEN {channel}')
+
+    def connect_paths(self, paths: List[Tuple[int, int]]) -> None:
+        """
+        to connect/close the specified channels.
+
+        Args:
+            paths: list of channels to connect [(r1, c1), (r2, c2), (r3, c3)]
+        """
+        channel_list_str = self.to_channel_list(paths)
```

Comment:
i think calls to `validate_value` should be extracted from `to_channel_list` and put for example here (in this method, similar to `connect`/`disconnect` methods), such that when implementing a driver for a different model of the switch matrix the developer will NOT need to rememeber to call `validate_value` when he makes a specific implementation of the `to_channel_list` method.

Reply:
Good point. I added the `validate_value` to all the connection-related methods, and removed it from the `to_channel_list`.